sdk/src/test/tutorial/java/com/finbourne/lusid/tutorials/ibor/ReferencePortfolio.java

```diff
@@ -0,0 +1,149 @@
+package com.finbourne.lusid.tutorials.ibor;
+
+import com.finbourne.features.LusidFeature;
+import com.finbourne.lusid.ApiClient;
+import com.finbourne.lusid.ApiException;
+import com.finbourne.lusid.api.InstrumentsApi;
+import com.finbourne.lusid.api.PortfoliosApi;
+import com.finbourne.lusid.api.ReferencePortfolioApi;
+import com.finbourne.lusid.model.*;
+import com.finbourne.lusid.utilities.*;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.*;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class ReferencePortfolio {
+    private static List<String> instrumentIds;
+    private static PortfoliosApi portfoliosApi;
+    private static ReferencePortfolioApi referencePortfoliosApi;
+
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        ApiConfiguration apiConfiguration = new ApiConfigurationBuilder().build(CredentialsSource.credentialsFile);
+        ApiClient apiClient = new ApiClientBuilder().build(apiConfiguration);
+
+        portfoliosApi = new PortfoliosApi(apiClient);
+        referencePortfoliosApi = new ReferencePortfolioApi(apiClient);
+        InstrumentsApi instrumentsApi = new InstrumentsApi(apiClient);
+
+        InstrumentLoader instrumentLoader = new InstrumentLoader(instrumentsApi);
+        instrumentIds = instrumentLoader.loadInstruments();
+
+        Collections.sort(instrumentIds);
+    }
+
+    @Test
+    @LusidFeature("F39")
+    public void create_reference_portfolio() throws ApiException {
+
+        // Declare name, code of the portfolio to create
+        String F39PortfolioCode = "F39J_ReferencePortfolioCode";
+        String F39PortfolioName = "F39J_Reference Portfolio name";
+
+        // Generate request using CreateReferencePortfolioRequest model
+        CreateReferencePortfolioRequest request = new CreateReferencePortfolioRequest()
+                .code(F39PortfolioCode)
+                .displayName(F39PortfolioName);
+
+        // Make the request using the reference portfolio API
+        Portfolio portfolio = referencePortfoliosApi.createReferencePortfolio(TestDataUtilities.TutorialScope, request);
+
+        // Test that result portfolio code is the same as request code
+        assertEquals(request.getCode(), portfolio.getId().getCode());
+
+        // Delete portfolio once the test is completed
+        portfoliosApi.deletePortfolio(TestDataUtilities.TutorialScope, F39PortfolioCode);
```

Comment:
Take a look at junit TearDown annotation to ensure this is called after a test even if it fails. 

The risk currently is imagine if your assertion fails. The test terminates before deletePortfolio is called. Because of the same portfolio name the next time you run the test and try to create a portfolio you'll likely get a portfolio already exists exception and terminate early. This will continue indefinitely.

But I'm glad your thinking about cleaning the test. I think there is a larger issue we have in SDK around not cleaning entities up properly. Let's bring this up with team separately.

Reply:
Makes sense, thanks @kkeenan-finbourne. @rizwansaeed -- what's your preference here? C# and Python SDKs already delete as part of each test to avoid the potential race condition with `tearDown`.